### ISSUE TITLE
Failed to process pragma: File Prims is not a valid option

### ISSUE THREAD

**Issue by cassiersg** [project-everest/everest#63]
When building the project (`./everest make`) on a Debian 10 machine, I get the following error:
```
make[1]: Entering directory '/home/cassiersg/everest/FStar/ulib'                                                                                                                                                   
/home/cassiersg/everest/FStar/bin/fstar.exe --use_extracted_interfaces true --warn_error -271 --use_hints --cache_dir .cache --hint_dir .cache --cache_checked_modules --odir _output LowStar.Endianness.fst       
LowStar.Endianness.fst(496,0-496,43): (Error 65) Failed to process pragma: File Prims is not a valid option                                                                                                        
1 error was reported (see above)                                                                                                                                                                                   
make[1]: *** [Makefile.verify:47: .cache/LowStar.Endianness.fst.checked] Error 1                                                                                                                                   
make[1]: Leaving directory '/home/cassiersg/everest/FStar/ulib'                                                                                                                                                    
make: *** [Makefile:10: all] Error 2                                                                                                                                                                               
make: Leaving directory '/home/cassiersg/everest/FStar/ulib'                                                                                                                                                       
================================================================================                                                                                                                                   
FAILURE: build failed for FStar
```
Any hint on how to solve this ?
Thanks.

**Comment by msprotz** [comment]
Could it be that you already have an `fstar.exe` somewhere in the PATH, or that you have incorrectly exported FSTAR_HOME?

**Comment by cassiersg** [comment]
I have the following (run in the same shell as the one where I got the previous error):
```
➜  everest git:(master) ✗ which fstar
fstar not found
➜  everest git:(master) ✗ which fstar.exe
/home/cassiersg/everest/FStar/bin/fstar.exe
➜  everest git:(master) ✗ echo $FSTAR_HOME
/home/cassiersg/everest/FStar
➜  everest git:(master) ✗ 
```

**Comment by s-zanella** [comment]
I assume you checked out the `master` branch and had run `./everest check` and `./everest pull`. 
Is it a fresh checkout or did you update from a previous build?

The error you get, as @msprotz hinted, suggests that `./everest make` didn't rebuild `fstar.exe`. Could it be that you've a running `fstar.exe` process and so overwriting the file fails? The easiest would be to retry after running `git clean -fdx` in the `FStar` directory.

**Comment by cassiersg** [comment]
I indeed took the master branch and run `./everest check` and `./everest pull`. This is a fresh checkout.
I did the `git clean -fdx` as you suggested, and re-running  `./everest -j 48 make`, I get now
```
/home/cassiersg/everest/FStar/bin/fstar.exe --use_extracted_interfaces true --warn_error -271 --use_hints --cache_dir .cache --hint_dir .cache --cache_checked_modules --odir _output legacy/FStar.HyperStack.IO.fst
FStar.ModifiesGen.fst(1494,0-1494,76): (Error 65) Failed to process pragma: File smt.qi.eager_threshold=100 is not a valid option
1 error was reported (see above)
make[1]: *** [Makefile.verify:47: .cache/FStar.ModifiesGen.fst.checked] Error 1
```

**Comment by msprotz** [comment]
What do you get with `/home/cassiersg/everest/FStar/bin/fstar.exe --version`?

**Comment by cassiersg** [comment]
```
➜  everest git:(master) ✗ /home/cassiersg/everest/FStar/bin/fstar.exe --version
F* 0.9.7.0-alpha1
platform=Linux_x86_64
compiler=OCaml 4.05.0
date=2020-01-06 06:16:20 +0000
commit=80f52718a0bbb9dec25c483974d5a2c63fd7bd83   
```

**Comment by msprotz** [comment]
```
jonathan@chartreuse:~/Code/fstar (master) $ git show 80f52718
fatal: ambiguous argument '80f52718': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```
are you on a custom commit?

**Comment by cassiersg** [comment]
No:
```
➜  everest git:(master) ✗ git show-ref HEAD                                    
2f5386cde98af5d55bb7bbaf42efc8cc10146c44 refs/remotes/origin/HEAD              
➜  everest git:(master) ✗ cd FStar
➜  FStar git:(master) git show-ref HEAD
80f52718a0bbb9dec25c483974d5a2c63fd7bd83 refs/remotes/origin/HEAD 
```
with
```
commit 80f52718a0bbb9dec25c483974d5a2c63fd7bd83 (HEAD -> master, origin/master, origin/HEAD)
Author: Dzomo the everest Yak <everbld@microsoft.com>                          
Date:   Mon Jan 6 06:16:20 2020 +0000

    [CI] regenerate hints + ocaml snapshot                                     
```

**Comment by omasanori** [comment]
I encountered the same problem yesterday and succeeded to build after pulling https://github.com/FStarLang/FStar/commit/a5943f6ff62f102196b6b885b5654205eb7c4831 into my workdir today.

**Comment by cassiersg** [comment]
This is fixed with the latest git (131cc2416c4).
